sys/linux/socket_netlink_generic_80211.txt

```diff
@@ -277,33 +405,27 @@ nl80211_sta_wme_policy [
 ] [varlen]
 
 nl80211_policy_mpath [
-	NL80211_ATTR_WIPHY		nlattr[NL80211_ATTR_WIPHY, nl80211_whipy]
-	NL80211_ATTR_IFINDEX		nlattr[NL80211_ATTR_IFINDEX, nl80211_ifindex]
-	NL80211_ATTR_WDEV		nlattr[NL80211_ATTR_WDEV, nl80211_wdev]
 	NL80211_ATTR_MAC		nlattr[NL80211_ATTR_MAC, ieee80211_mac_addr]
 	NL80211_ATTR_MPATH_NEXT_HOP	nlattr[NL80211_ATTR_MPATH_NEXT_HOP, ieee80211_mac_addr]
 ] [varlen]
 
 nl80211_policy_bss [
-	NL80211_ATTR_WIPHY			nlattr[NL80211_ATTR_WIPHY, nl80211_whipy]
-	NL80211_ATTR_IFINDEX			nlattr[NL80211_ATTR_IFINDEX, nl80211_ifindex]
-	NL80211_ATTR_WDEV			nlattr[NL80211_ATTR_WDEV, nl80211_wdev]
 	NL80211_ATTR_BSS_CTS_PROT		nlattr[NL80211_ATTR_BSS_CTS_PROT, int8]
 	NL80211_ATTR_BSS_SHORT_PREAMBLE		nlattr[NL80211_ATTR_BSS_SHORT_PREAMBLE, int8]
 	NL80211_ATTR_BSS_SHORT_SLOT_TIME	nlattr[NL80211_ATTR_BSS_SHORT_SLOT_TIME, int8]
 	NL80211_ATTR_BSS_HT_OPMODE		nlattr[NL80211_ATTR_BSS_HT_OPMODE, int16]
 	NL80211_ATTR_AP_ISOLATE			nlattr[NL80211_ATTR_AP_ISOLATE, int8]
 	NL80211_ATTR_P2P_CTWINDOW		nlattr[NL80211_ATTR_P2P_CTWINDOW, int8]
 	NL80211_ATTR_P2P_OPPPS			nlattr[NL80211_ATTR_P2P_OPPPS, bool8]
-	NL80211_STA_WME_MAX_SP			nlattr[NL80211_STA_WME_MAX_SP, array[int8, 0:NL80211_MAX_SUPP_RATES]]
+	NL80211_STA_WME_MAX_SP			nlattr[NL80211_STA_WME_MAX_SP, nl80211_rates]
 ] [varlen]
 
 nl80211_policy_reg [
 	NL80211_ATTR_WIPHY		nlattr[NL80211_ATTR_WIPHY, nl80211_whipy]
 	NL80211_ATTR_USER_REG_HINT_TYPE	nlattr[NL80211_ATTR_USER_REG_HINT_TYPE, flags[nl80211_user_reg_hint_type, int32]]
 	NL80211_ATTR_SOCKET_OWNER	nlattr[NL80211_ATTR_SOCKET_OWNER, void]
 	NL80211_ATTR_REG_ALPHA2		nlattr[NL80211_ATTR_REG_ALPHA2, string[nl80211_reg_alpha2]]
-	NL80211_ATTR_REG_RULES		nlnest[NL80211_ATTR_REG_RULES, array[reg_rule_policy]]
+	NL80211_ATTR_REG_RULES		nlnest[NL80211_ATTR_REG_RULES, array[nlnest[0, array[reg_rule_policy]]]]
```

Comment:
Why do we need this double layer of `nlnest` here and below?

Reply:
Because it expects an array of entities that contain attributes themselves. For example, the code that parses `NL80211_ATTR_REG_RULES` is [here](https://elixir.bootlin.com/linux/latest/source/net/wireless/nl80211.c#L7495). `nla_for_each_nested` loops through each attribute that is nested in `NL80211_ATTR_REG_RULES`, then `nla_parse_nested_deprecated` parses attributes that belong to each of those nested attributes.

If there was only a `nla_parse_nested_deprecated` call, then double nesting would be unnecessary.